pulp_rpm/app/tasks/publishing.py

```diff
@@ -231,7 +231,7 @@ def handle_sub_repos(self, distribution_tree):
                 repository = addon_or_variant.repository.cast()
                 repository_version = repository.latest_version()
 
-                if repository_version and repository.sub_repo:
+                if repository_version and repository.user_hidden:
```

Comment:
To backport, we could drop the migration and make this line "...and (repository.user_hidden or repository.sub_repo):".  We would stop adding more sub_repos, the new code would be used, and when a user upgrades, **then** the migration runs and removes the sub_repo field.

Add a comment to that effect here, to help whoever gets the task down the line.

Reply:
I don't disagree - and once we get past "no backports with a migration", we're good. Objection withdrawn :)

Reply:
@ggainey Just to clarify, is this is an objection to backporting as-is with the migration, or a request to have a discussion about what our policies are before doing so, or neither of those?

Reply:
This is "if our backport process allows migrations" (which it does not, currently), then we're fine. If we continue to not-allow migrations in backports, but there was pressure to to so in this instance, there is a relatively-straightforward path to do so. My comment was more to make the life of a future-cherrypicker easier in that case.

Reply:
I may bring it up at open floor since this is something we may eventually end up doing (not necessarily now).

In the meantime should I plan to just fix the cache invalidation on the 3.14 branch?

Reply:
@dralley i would probably ask stakerholder who's on 3.14 whether this is something they really want to have, in that case I'd proceed with the modified patch since it is a straightforward one. I would not adjust master code to satisfy needs for a potential backport.